vsp.go

```diff
@@ -57,65 +61,315 @@ func (mw *MultiWallet) NewVSPClient(vspHost string, walletID int, purchaseAccoun
 		ticketsToFees: make(map[chainhash.Hash]PendingFee),
 	}
 
-	ctx, cancel := mw.contextWithShutdownCancel()
-	mw.cancelFuncs = append(mw.cancelFuncs, cancel)
+	ctx, _ := mw.contextWithShutdownCancel()
 
-	v.vspClient = newVSPClient(vspHost, nil, sourceWallet.Internal())
-	vspInfo, err := v.GetInfo(ctx)
+	vsp.client = newVSPClient(vspHost, nil, sourceWallet.Internal())
+	vspInfo, err := vsp.GetInfo(ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	v.ctx = ctx
```

Comment:
`vsp.ctx` is no longer set, causes a crash when purchasing tickets from godcr #711.